### PR TITLE
Sequence Expansion - Add Channel and Parameter Dictionaries support

### LINKED ISSUE #1439: Sequence Expansion with Channel and Parameter Dictionaries
### Checked for duplicates

No - I haven't checked

### Alternatives considered

No - I haven't considered

### Related problems

_No response_

### Describe the feature request

The new user editor uses Channel and Parameter dictionaries. We want to use this on the expansion side as well. 

- [ ] Convert expansion rules to using parcels 
- [ ] Convert expansion sets to using parcels 
- [ ] Support uploading the 3 types of dictionaries Command, Channel, Parameters
- [ ] Typecheck against Channel and Parameter dictionary types
- [ ] Support command expansion and sequence generation with the Channel and Parameter Dictionary. 
- [ ] Update e2e test to support parcels

### PR DESCRIPTION
**Pull request by goetzrrGit**
* **Tickets addressed:** closes #1439
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
Big refactor and changes to support channel and parameter dictionaries.

Frontend PR: [Coming soon](https://github.com/NASA-AMMOS/aerie-ui/pull/1273) 

- [x] Convert expansion rules to using parcels 
- [x] Convert expansion sets to using parcels 
- [x] Support uploading the 3 types of dictionaries Command, Channel, Parameters
- [x] Typecheck against Channel and Parameter dictionary types
- [x] Support command expansion and sequence generation with the Channel and Parameter Dictionary. 
- [x] Update e2e test to support parcels

## Verification
Manually ran expansion by hand  as well as the e2e test

### CODE REVIEW

File: deployment/hasura/metadata/actions.graphql
Comment (cohansen) [inline deployment/hasura/metadata/actions.graphql:222]: There's still `getCommandTypeScript`, `getUserSequenceSeqJson`, and `GetUserSequenceSeqJsonBulkInput` that are tied to `commandDictionaryID`. Do those need to be updated as well?

File: sequencing-server/package.json
Comment (cohansen) [inline sequencing-server/package.json:14]: I forgot, are the `.ts` files here just wrappers for the JSON objects?

File: sequencing-server/sql/sequencing/tables/channel_dictionary.sql
Comment (cohansen) [inline sequencing-server/sql/sequencing/tables/channel_dictionary.sql]: I know we spoke about this but leaving a note, we can probably remove the `parsed_json` field from all of these tables and just pull the files directly.

File: sequencing-server/sql/sequencing/tables/command_dictionary.sql
Comment (cohansen) [inline sequencing-server/sql/sequencing/tables/command_dictionary.sql]: Ditto.

File: sequencing-server/sql/sequencing/tables/parameter_dictionary.sql
Comment (cohansen) [inline sequencing-server/sql/sequencing/tables/parameter_dictionary.sql]: Ditto.

File: sequencing-server/src/lib/codegen/CommandTypeCodegen.ts
Comment (cohansen) [inline sequencing-server/src/lib/codegen/CommandTypeCodegen.ts]: If we're going with this standard we should probably tie these values to a real `enum` rather than just using magic strings.

File: sequencing-server/src/types/ChannelTypes.ts
Comment (cohansen) [inline sequencing-server/src/types/ChannelTypes.ts:1]: Do these types need to stay in sync with the types exported from the AMPCS parsing library?

File: sequencing-server/src/types/ParameterTypes.ts
Comment (cohansen) [inline sequencing-server/src/types/ParameterTypes.ts:1]: Do these types need to stay in sync with the types exported from the AMPCS parsing library?

File: sequencing-server/src/app.ts
Comment (cohansen) [inline sequencing-server/src/app.ts]: Same as the other `enum` comment.

**Review by cohansen — CHANGES_REQUESTED** [review]
Some minor comments / questions.

File: deployment/hasura/metadata/actions.graphql
Comment (goetzrrGit) [reply deployment/hasura/metadata/actions.graphql:222]: `getCommandTypeScript` We can leave it as this because this file is generated by the command dictionary.

I will probably remove`getUserSequenceSeqJson` `GetUserSequenceSeqJsonBulkInput` as you can use hasura to fetch the user seqjson

File: sequencing-server/package.json
Comment (goetzrrGit) [reply sequencing-server/package.json:14]: These are the type definition files for Channel and Parameters.

File: sequencing-server/sql/sequencing/tables/channel_dictionary.sql
Comment (goetzrrGit) [reply sequencing-server/sql/sequencing/tables/channel_dictionary.sql]: Could I add a todo comment. As I know it will require more frontend changes which we can do later after we merge into develop

Comment (cohansen) [reply sequencing-server/sql/sequencing/tables/channel_dictionary.sql]: Yeah 100%, don't want to hold up the merge.

File: sequencing-server/src/lib/codegen/CommandTypeCodegen.ts
Comment (goetzrrGit) [reply sequencing-server/src/lib/codegen/CommandTypeCodegen.ts]: Good idea, I created one.

File: sequencing-server/src/types/ChannelTypes.ts
Comment (goetzrrGit) [reply sequencing-server/src/types/ChannelTypes.ts:1]: Yes, they do, I don't know of an elegant solution to this. Chet is aware of this.

File: sequencing-server/src/types/ParameterTypes.ts
Comment (goetzrrGit) [reply sequencing-server/src/types/ParameterTypes.ts:1]: Yes, they do, I don't know of an elegant solution to this. Chet is aware of this.

File: sequencing-server/src/app.ts
Comment (goetzrrGit) [reply sequencing-server/src/app.ts]: Good idea, I created one.

**Review by cohansen — APPROVED** [review]

**Pull request merged**
